File: _inc/laravel/app/Http/Middleware/PusherConfig.php
<?php

namespace App\Http\Middleware;

use Closure;
use App\Config\Constants\SettingsConstants;
use App\Models\Utility;
use Illuminate\{Http\Request, Support\Facades\Log};
use Symfony\Component\HttpFoundation\Response;

final class PusherConfig
{
    /**
     * Apply dynamic Pusher configuration based on stored settings.
     *
     * @param  Request   $request
     * @param  \Closure  $next
     * @return Response
     */
    public function handle(Request $request, Closure $next): Response
    {
        try {
            $settings = Utility::settingsById(1);
            if (is_array($settings) && $settings) {
                foreach (
                    [
                        'chatify.pusher.key'             => 'pusher_app_key',
                        'chatify.pusher.secret'          => 'pusher_app_secret',
                        'chatify.pusher.app_id'          => 'pusher_app_id',
                        'chatify.pusher.options.cluster' => 'pusher_app_cluster',
                    ] as $configKey => $settingKey
                ) {
                    config([$configKey => $settings[$settingKey] ?? null]);
                }
            }
            return $next($request);
        } catch (\Throwable $e) {
            Log::warning('PusherConfig: failed to load settings', [
                'message' => $e->getMessage(),
            ]);
            Log::channel(SettingsConstants::ERR_TRACE)->debug('PusherConfig trace', [
                'trace' => $e->getTraceAsString(),
            ]);
            throw $e;
        }
    }
}

File: _inc/_DEPRECATED_django/erp_prestech_backend/app/Http/Middlewares/pusher_config.py
import logging
from django.http import HttpRequest
from django.utils.deprecation import MiddlewareMixin
from ...Models.utils.utility import Utility

logger = logging.getLogger(__name__)

class PusherConfig(MiddlewareMixin):
  def process_request(self, request: HttpRequest) -> None:
    try:
      settings_dict = Utility.settings_by_id(1)
      if settings_dict:
        request.pusher_config = {
          'key': settings_dict.get('pusher_app_key', ''),
          'secret': settings_dict.get('pusher_app_secret', ''),
          'app_id': settings_dict.get('pusher_app_id', ''),
          'cluster': settings_dict.get('pusher_app_cluster', ''),
        }
    except Exception as e:
      logger.error("Failed in PusherConfig.process_request: %s", e)
